pyEpiabm/pyEpiabm/core/microcell.py

```diff
@@ -41,6 +41,7 @@ def __init__(self, cell):
         self.compartment_counter = _CompartmentCounter(
             f"Microcell {id(self)}")
         self.closure_start_time = None
+        self.distancing_start_time = None
```

Comment:
I wonder how many of these attributes you think there will be - we already have two related to interventions and I am not sure how many more we will need. If its a lot, maybe this isn't the best way to set this up? 

For example, could we consider not specifying these in the init file, and creating them on the fly if required? It would reduce the memory requirement for example

Reply:
There are four start_time (each intervention has one). I think we have to initialise it rather than creating it on the fly because we are checking for example if 'microcell.distancing_start_time is not None' each time (e.g. in social_distancing.py). If we are not initialising it from the beginning, it contributes to attribute error.

Reply:
That makes sense, but its very easy to check if an attribute exists (`hasattr()`) which could be added to the if logic.

Reply:
Yeah, I removed initialisation for four interventions in person.py and microcell.py. Now I check if the start_time exists with hasattr() before checking if its value is None.

Reply:
Okay! We indeed still need  to check if it is None because it is set to None after the policy is inactivated. Do the tests still run? As we are checking often if the value is None. Or did you change that as well?

Reply:
I've changed for all four interventions and conflicts will be solved when merging!